### DISCUSSION TITLE
Help with Websocket

### DISCUSSION THREAD

**Discussion by jainkv107** [chatwoot/chatwoot#4967] [Chatwoot APIs]
Hello - I am facing issue with WS and I am not able to get presence or messages on WebSocket. I was able to successfully bring up chatwoot version 2.5, create account, inboxes, users and contacts.  

I am able to connect to ws://localhost:3000/cable.  I am also able to subscribe 
{"command":"subscribe","identifier":"{\"channel\":\"RoomChannel\",\"pubsub_token\":\"J4mKb7X1GawVMQXkhqTcbBDX\",\"account_id\":1}"} 

I get back {"identifier":"{\"channel\":\"RoomChannel\",\"pubsub_token\":\"J4mKb7X1GawVMQXkhqTcbBDX\",\"account_id\":1}","type":"confirm_subscription"}.  

I see regular ping message {"type":"ping","message":1656891273}

Please advice.

Thanks,
Vik

**Comment by sojan-official** [comment]
please refer https://www.chatwoot.com/docs/product/others/websocket-events/#publishing-presence-to-the-websocket-server

**Comment by jainkv107** [comment]
Hi Sojan - I looked at the link above.  In my installation, I am not receiving the messages (message.created etc) from Chatwoot even after successful subscription.   With the same code, I am able to connect to chatwoot on wss://app.chatwoot.com/cable.   I used default helm chart to install.  Even when new messages are posted, I don't see it in websocket.  If I refresh the site as an agent, I see them posted.  Please help.

I am able to connect to ws://localhost:3000/cable. I am also able to subscribe
{"command":"subscribe","identifier":"{"channel":"RoomChannel","pubsub_token":"J4mKb7X1GawVMQXkhqTcbBDX","account_id":1}"}

I get back {"identifier":"{"channel":"RoomChannel","pubsub_token":"J4mKb7X1GawVMQXkhqTcbBDX","account_id":1}","type":"confirm_subscription"}.

I see regular ping message {"type":"ping","message":1656891273}

**Reply by sojan-official** [reply]
@jainkv107  if you are trying to connect as an agent, you also have to pass the user_id of the agent in the subscribe command. 

```
connection.send(
  stringify({
    command: "subscribe",
    identifier: stringify({
      channel: "RoomChannel",
      pubsub_token: pubSubToken,
      account_id: accountId,
      user_id: userId,
    }),
  })
);
```
